rollingpin/frontends.py

```diff
@@ -160,6 +161,28 @@ def on_deploy_end(self):
         elapsed = time.time() - self.start_time
         print "*** elapsed time: %d seconds" % elapsed
 
+        # TODO: Make this smarter.  It should only show anomolous
+        # stuff.  Or maybe color that stuff differently.
+        #
+        # TODO: Check if component report was even requested before going
+        # through them all?
+        report = collections.defaultdict(lambda: collections.Counter())
+        for host, results in self.host_results.iteritems():
+            # Messed up hosts won't have results
+            if 'results' not in results:
+                continue
+            for command, output in results['results']:
+                if command[0] != 'component_report':
+                    continue
+                for component, sha in output['components'].iteritems():
+                    report[component][sha] += 1
+        if report:
+            print colorize("*** component report", Color.BOLD(Color.GREEN))
+            print "COMPONENT\tSHA\tCOUNT"
+            for component in report.keys():
+                for sha, count in report[component].iteritems():
+                    print "%s\t%s\t%s" % (component, sha, count)
```

Comment:
Meaning making the output look nicer?  Those are tabs in between, so it ends up being split into columns pretty well right now (though I don't know if it'll still work well with longer outputs).